### PR TITLE
IBP-4731 brapi sync study

### PR DESCRIPTION
**Pull request by nahuel-soldevilla**

### CODE REVIEW

File: brapi-sync.iml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
+    <content url="file://$MODULE_DIR$" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+</module>
```
Comment (nahuel-soldevilla) [inline brapi-sync.iml:8]: I forgot to add iml to .gitignore, can you add please

File: angular.json [hunk 2]
```diff
@@ -1,5 +1,8 @@
 {
   "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
+  "cli": {
+    "analytics": "4c45c6ba-6cab-42db-bda7-793a0a0ce330"
+  },
```
Comment (nahuel-soldevilla) [inline angular.json]: was this automatically added by angular cli? weird, we should probably disable it

File: src/app/app.module.ts [hunk 3]
```diff
@@ -39,4 +58,11 @@ import { AuthInterceptor } from './auth.interceptor';
   bootstrap: [AppComponent]
 })
 export class AppModule {
+
+  constructor(faIconLibrary: FaIconLibrary) {
+    // Add an icon to the library for convenient access in other components
+    // Only add specific icons explicitly to avoid loading all icons to the bundle.
+    faIconLibrary.addIcons(faCaretDown, faCaretRight, faPlus, faCheck, faTimes);
+  }
```
Comment (nahuel-soldevilla) [inline src/app/app.module.ts:66]: :+1: 
we can improve the germplasm study button now. It should probably work with bootstrap `input-group` too

File: package.json [hunk 4]
```diff
@@ -12,9 +12,15 @@
...
     "@angular/platform-browser-dynamic": "~11.2.4",
     "@angular/router": "~11.2.4",
+    "@fortawesome/angular-fontawesome": "^0.8.2",
+    "@fortawesome/fontawesome-svg-core": "^1.2.34",
+    "@fortawesome/free-regular-svg-icons": "^5.15.3",
+    "@fortawesome/free-solid-svg-icons": "^5.15.3",
     "@ng-bootstrap/ng-bootstrap": "^9.0.2",
-    "@solgenomics/brapijs": "git+ssh://git@github.com:solgenomics/BrAPI-js.git#semver:^1.0.1",
+    "@ng-select/ng-select": "6.1.0",
+    "@solgenomics/brapijs": "git+https://github.com/solgenomics/BrAPI.js.git",
```
Comment (nahuel-soldevilla) [inline package.json]: technically, I could be dangerous not to specify the version, there could be breaking changes..

File: package.json [hunk 5]
```diff
@@ -12,9 +12,15 @@
     "@angular/platform-browser": "~11.2.4",
     "@angular/platform-browser-dynamic": "~11.2.4",
     "@angular/router": "~11.2.4",
+    "@fortawesome/angular-fontawesome": "^0.8.2",
+    "@fortawesome/fontawesome-svg-core": "^1.2.34",
+    "@fortawesome/free-regular-svg-icons": "^5.15.3",
+    "@fortawesome/free-solid-svg-icons": "^5.15.3",
     "@ng-bootstrap/ng-bootstrap": "^9.0.2",
-    "@solgenomics/brapijs": "git+ssh://git@github.com:solgenomics/BrAPI-js.git#semver:^1.0.1",
+    "@ng-select/ng-select": "6.1.0",
```
Comment (nahuel-soldevilla) [inline package.json:20]: Nice, this one seems like it's much more supported than the alternatives, I don't know why I did't find it last time I wanted something similar..

File: src/styles.css [hunk 6]
```diff
@@ -1,2 +1,28 @@
 /* You can add global styles to this file, and also import other style files */
 @import "~bootstrap/dist/css/bootstrap.css";
+@import "~@ng-select/ng-select/themes/default.theme.css";
+
+.btn-primary {
+  color: #212529;
+  background-color: #ff8109;
+  border-color: #ff8109;
+}
```
Comment (nahuel-soldevilla) [inline src/styles.css:9]: This is fine but we probably don't want to go too overboard with customizations, implementors could/will want to apply their own style

**Review by nahuel-soldevilla — COMMENTED** [review]
Awesome.
I don't think there's any blocker for merge, though it'll be nice to do some update in the readme to list the new capabilities.
After merge, given that there are a lot new features and probably some breaking changes (?), we can bump to v2 using `npm version major` (we are not using the proper version convention so probably we should mention also in the readme that this is still an alpha version)

File: brapi-sync.iml (hunk 1 above)
Comment (abatac) [reply brapi-sync.iml:8]: Done

File: angular.json (hunk 2 above)
Comment (abatac) [reply angular.json]: Yup, this was added by angular cli. I just removed this code.

File: package.json (hunk 4 above)
Comment (abatac) [reply package.json]: Done

**Pull request merged**